### ISSUE TITLE
Spec is unclear about how much of the key response needs to be signed

### ISSUE THREAD

**Issue by turt2live** [matrix-org/matrix-doc#2194]
In practice synapse includes a `tls_fingerprints` (as carryover from a legacy spec) however it is not described as needing covering by the signing algorithm. We should spell out for notaries that they need to verify signature and sign the whole object, non-standard keys and all.

**Comment by richvdh** [comment]
yeees, although they just follow the standard [Signing JSON](https://matrix.org/docs/spec/appendices#signing-json) algorithm...

**Comment by turt2live** [comment]
True, but it's a bit of a trap that notaries have to be designed to store arbitrary JSON rather than well-defined JSON it can pick apart in a database. Just needs a small note to say that other fields might be present to aid development, I think.
